### ISSUE TITLE
`UntypedObjectDeserializer` mixes multiple unwrapped collections (related to #2733)

### ISSUE THREAD

**Issue by Migwel** [FasterXML/jackson-databind#3025]
(needed for https://github.com/FasterXML/jackson-dataformat-xml/issues/445)

There's an issue when deserializing XML nodes containing multiple unwrapped collections. For example:

```
<person>
    <name>a</name>
    <name>b</name>
    <surname>c</surname>
    <surname>d</surname>
</person>
```
gets deserialized into
```
{
    name = [ c, d ], 
    surname = d
}
```

Expected deserialization:
```
{
    name = [ a, b ], 
    surname = [c, d]
}
```

**Comment by cowtowncoder** [comment]
Note: relevant for #2733 that was added in 2.12.0.

**Comment by cowtowncoder** [comment]
Fixed for 2.12.2.
